flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalLookupJoin.scala

```diff
@@ -42,15 +43,22 @@ class BatchPhysicalLookupJoin(
     temporalTable: RelOptTable,
     tableCalcProgram: Option[RexProgram],
     joinInfo: JoinInfo,
-    joinType: JoinRelType)
+    joinType: JoinRelType,
+    lookupHint: Option[RelHint] = Option.empty,
+    enableLookupShuffle: Boolean = false,
+    preferCustomShuffle: Boolean = false)
```

Comment:
Would it be better to rename both "LookupShuffle" and "CustomShuffle" to "CustomLookupShuffle"?

Besides, it might be better to clarify the difference between "prefer", "support" and "enable".

Reply:
enableLookupShuffle means user tend to shuffle input stream, preferCustomShuffle means dim table prefer custom partitioners to hash partitioner. These are two concepts that are not exactly equivalent.

Reply:
For custom shuffle, `prefer` is used instead of `enable` because the connector may implement interface `SupportsLookupCustomShuffle`, but sometimes it does not want to shuffle data(i.e. SupportsLookupCustomShuffle#getPartitioner return EMPTY).